cli/azd/pkg/infra/provisioning/bicep_provider.go

```diff
@@ -186,21 +195,22 @@ func (p *BicepProvider) Deploy(ctx context.Context, deployment *Deployment, scop
 
 			// Start the deployment
 			modulePath := p.modulePath()
-			parametersFilePath := p.parametersFilePath()
-			deployResult, err := p.deployModule(ctx, scope, modulePath, parametersFilePath)
+			bicepDeploymentData := pd.Details.(BicepDeploymentDetails)
```

Comment:
We should validate the type assertion.
```suggestion
			bicepDeploymentData, ok := pd.Details.(BicepDeploymentDetails)
			if !ok {
			  // Some error handling
			}
```

Reply:
I omitted that on purpose--if the details are not present, or are wrong type, it means we do not have the path to parameters file. It is an unrecoverable coding bug and deployment cannot proceed. Panic is fine here.